rust/core-lib/src/file.rs

```diff
@@ -260,10 +260,14 @@ impl CharacterEncoding {
     }
 }
 
-/// Returns the modification timestamp for the file at a given path,
-/// if present.
-fn get_mod_time<P: AsRef<Path>>(path: P) -> Option<SystemTime> {
-    File::open(path).and_then(|f| f.metadata()).and_then(|meta| meta.modified()).ok()
+/// Returns the SHA256 checksum of a file at a given path, if present.
+fn get_sha256<P: AsRef<Path>>(path: P) -> Option<String> {
+    let mut hasher = Sha256::new();
+    File::open(path)
+        .and_then(|mut f| std::io::copy(&mut f, &mut hasher))
+        .map(|_| hasher.result())
+        .map(|s| format!("{:x}", s))
```

Comment:
Why would you hex encode it? Rust works very well with binary data.

Reply:
Ah, was mainly just so I can print out the sha when it has changed, not quite sure if we want to do rhat though.

Reply:
If we want to do that, then let's add an implementation of the [Debug](https://doc.rust-lang.org/std/fmt/trait.Debug.html) trait to hex encode when printing.

Reply:
Alright, I've just opted to not do that since we don't print it out anywhere yet.

Reply:
Converting `Vec` is also an unnecessary inefficiency. Better store the `GenericArray` you get from the `sha2` crate, which doesn't involve any pointers, lengths or allocations.

Reply:
Alright, but to be honest I'm not exactly sure how we return a `GenericArray<u8, ArrayLength<u8>>` (since ArrayLength only is a trait and the generic-array docs haven't helped me and the sha2 docs also uses a Vec for this)

Reply:
It's `generic_array::GenericArray<u8, generic_array::typenum::consts::U32>`. It's also possible to use `[u8; 32]` directly without wrapping it in `GenericArray`, which is stored exactly the same way in memory, but then it won't be compatible with generic functions that require a generic array. Hopefully this distinction will eventually go away as [const generics](https://rust-lang.github.io/rfcs/2000-const-generics.html) are implemented.

Reply:
For our purposes I think that `[u8; 32]` is fine. I also don't really mind the vec, since this is being computed rarely and the performance impact will be ~nil.